src/Microsoft.AspNetCore.Http/Extensions/HttpRequestRewindUtility.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http.Internal;
+
+namespace Microsoft.AspNetCore.Http.Extensions
+{
+    public static class HttpRequestRewindUtility
```

Comment:
Should we have a DI service instead? Since it does write to the disk sometimes, it could be surprising for a test. A test should be able to mock this. 
Thinking more this is probably not a problem as during test a request body is usually a seekable stream, in which case we don't write to disk anyways.